### PR TITLE
odashynych: added test to cover sale sort

### PR DESCRIPTION
**Pull request by clarmmym**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java [hunk 1]
```diff
@@ -63,4 +64,9 @@ public SearchResultsPage sortBy(ProductSort sort) {
         $x(String.format("//rz-sort//select//option[contains(@value, '%s')]", sort.getOptionXpath())).click();
         return this;
     }
+
+    public boolean isOnSale(int productNumber) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/SearchResultsPage.java:68]: this should be in Product class

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.softserveinc.ita.rozetka;
+
+import com.softserveinc.ita.rozetka.data.ProductFilter;
+import com.softserveinc.ita.rozetka.data.ProductSort;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+public class FilterProductTest extends TestRunner {
+    @Test
+    public void saleSortTest() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:10]: ???

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                .search(searchItem)
+                .sortBy(ProductSort.PROMOTION);
+        SoftAssert softAssert = new SoftAssert();
+
+        softAssert.assertTrue(searchResultsPage.isOnSale(1));
+        softAssert.assertTrue(searchResultsPage.isOnSale(15));
+        softAssert.assertTrue(searchResultsPage.isOnSale(60));
+
+        searchResultsPage = searchResultsPage
+                .getHeader()
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:23]: duplication

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    public void verifySaleSortTest() {
+        Header header = homePage.getHeader();
+        SearchResultsPage searchResultsPage = header
+                .search("laptop")
+                .sortBy(ProductSort.PROMOTION);
+
+        SoftAssert softAssert = new SoftAssert();
+
+        softAssert.assertTrue(searchResultsPage.getProduct(1).isOnSale());
+        softAssert.assertTrue(searchResultsPage.getProduct(59).isOnSale());
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:20]: what if in the middle of products there's a bug?

Comment (toleksyn) [reply src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:20]: what if there's less than 59 products?

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+        searchResultsPage = searchResultsPage
+                .sortBy(ProductSort.RATING)
+                .getFilter()
+                .filter(ProductFilter.PROMOTION);
+
+        softAssert.assertTrue(searchResultsPage.getProduct(2).isOnSale());
+        softAssert.assertTrue(searchResultsPage.getProduct(60).isOnSale());
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:28]: same

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 6]
```diff
@@ -9,6 +14,54 @@
 import static com.softserveinc.ita.rozetka.data.ProductFilter.*;
 
 public class FilterProductTest extends TestRunner {
+    @Test
+    public void verifySaleSort() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:18]: rename

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 7]
```diff
@@ -9,6 +14,54 @@
...
+        Header header = homePage.getHeader();
+        SearchResultsPage searchResultsPage = header
+                .search("laptop")
+                .sortBy(ProductSort.PROMOTION);
+
+        Assert.assertTrue(searchResultsPage.getProductsSize() >= 60, "There aren't 60 products");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:24]: assertj

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 8]
```diff
@@ -9,6 +14,54 @@
...
+        SoftAssertions softAssertions = new SoftAssertions();
+        String errorMessage = "Product with number \"%s\" isn't on sale";
+        int productNumber = 1;
+        softAssertions.assertThat(searchResultsPage.getProduct(productNumber).isOnSale())
+                .withFailMessage(String.format(errorMessage, productNumber))
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:29]: .as()

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 9]
```diff
@@ -9,6 +14,54 @@
...
+                .isTrue();
+
+        productNumber = 25;
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:32]: inline these variables

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java (hunk 6 above)
Comment (dankomax) [reply src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:18]: veryfyUserCanSortProductsByPromotion

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 10]
```diff
@@ -12,6 +15,48 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class FilterProductTest extends TestRunner {
+    @Test
+    public void verifyProductsSortingByPromotion() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:19]: is it sorting or filtering?

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 11]
```diff
@@ -12,6 +15,48 @@
...
+                    .getProduct(productNumber)
+                    .isOnSale();
+
+            softAssertions.assertThat(isProductOnSale)
+                    .as(errorMessage, productNumber)
+                    .isTrue();
+        }
+
+        searchResultsPage = searchResultsPage
+                .sortBy(ProductSort.RATING)
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:43]: is it sorting by promotion or rating?

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 12]
```diff
@@ -13,6 +15,33 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class FilterProductTest extends TestRunner {
+    @Test
+    public void verifyProductsSaleFilter() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:19]: rename

File: src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java [hunk 13]
```diff
@@ -13,6 +15,33 @@
...
+                .filter(ProductFilter.PROMOTION);
+
+        SoftAssertions softAssertions = new SoftAssertions();
+        for (int productNumber : new int[]{2, 40, 60}) {
+            boolean isProductOnSale = searchResultsPage
+                    .getProduct(productNumber)
+                    .isOnSale();
+
+            softAssertions.assertThat(isProductOnSale)
+                    .as("Product with number %s should be on sale", productNumber)
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/FilterProductTest.java:38]: number + " product should ... "

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka/SortingProductTest.java [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+        assertThat(searchResultsPage.getProductsQuantity())
+                .as("Product quantity should be sufficient")
+                .isGreaterThanOrEqualTo(60);
+
+        searchResultsPage = searchResultsPage.sortBy(ProductSort.PROMOTION);
+
+        SoftAssertions softAssertions = new SoftAssertions();
+
+        for (int productNumber : new int[]{1, 25, 59}) {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka/SortingProductTest.java:25]: why it's {1, 25, 59} and in the previous test it's {2, 40, 60} ?

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Pull request merged**